packages/notification/channel/mailgun.ts

```diff
@@ -42,17 +49,70 @@ export const validator = Joi.object().keys({
     .label('Mailgun Recipients'),
   apiKey: Joi.string().required().label('Mailgun API Key'),
   domain: Joi.string().required().label('Mailgun Domain'),
-  username: Joi.string().label('Mailgun Username'),
 })
 
+const isContent = (data: NotificationMessage | Content): data is Content => {
+  const keys: Array<keyof Content> = ['from', 'subject', 'html', 'text']
+
+  return keys.every((k) => Object.prototype.hasOwnProperty.call(data, k))
+}
+
+const _send = async (
+  { apiKey, domain, recipients }: NotificationData,
+  data: NotificationMessage | Content
+): Promise<void> => {
+  const url = `https://api.mailgun.net/v3/${domain}/messages`
+  const auth = Buffer.from('api:' + apiKey).toString('base64')
+  const theData = isContent(data)
+    ? data
+    : getContent({ body: data.body, subject: data.subject, domain })
+
+  const formData = new FormData()
+  formData.append('from', theData.from)
+  formData.append('subject', theData.subject)
+  formData.append('html', theData.html)
+  formData.append('text', theData.text)
+  for (const email of recipients) formData.append('to', email)
+  if (theData.headerReference)
+    formData.append('h:References', theData.headerReference)
+
+  await sendHttpRequest({
+    method: 'POST',
+    url,
+    headers: {
+      Authorization: 'Basic ' + auth,
+    },
+    data: formData,
+  })
+}
+
 export const send = async (
-  { apiKey: key, domain, recipients, username = 'api' }: NotificationData,
-  { body, subject }: NotificationMessage
+  notificationData: NotificationData,
+  notificationMessage: NotificationMessage
+): Promise<void> => {
+  await _send(notificationData, notificationMessage)
+}
+
+export const sendWithCustomContent = async (
```

Comment:
Could you elaborate why it needs to be exported?

Reply:
Need to be exported to be used in neosense. Also, if not exported it will get error typescript 
```
sendWithCustomContent is declared but its value is never read.
```